crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -646,16 +661,18 @@ async fn bulk_operations(
     let mut response = Vec::<ContextBulkResponse>::new();
     conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
         for action in reqs.into_inner().into_iter() {
+            let tenant = tenant.clone();
             match action {
                 ContextAction::Put(put_req) => {
-                    let put_resp = put(Json(put_req), transaction_conn, true, &user)
-                        .map_err(|err| {
-                            log::error!(
-                                "Failed at insert into contexts due to {:?}",
+                    let put_resp =
+                        put(Json(put_req), transaction_conn, true, &user, &state, tenant)
```

Comment:
we can rather accept a reference of the tenant in the `put` function, then you can avoid cloning tenant